packages/core/content-releases/server/src/migrations/database/5.0.0-document-id-in-actions.ts

```diff
@@ -0,0 +1,33 @@
+import type { Migration } from '@strapi/database';
+
+/**
+ * On v4, release actions are linked with entries using the built in Polymorphic relations.
+ *
+ * On v5, we are going to save entryDocumentId on the release action and make the link manually.
+ * This because entryId is not a reliable way to link documents, as it can change.
+ */
+export const addEntryDocumentToReleaseActions: Migration = {
+  name: 'content-releases::5.0.0-add-entry-document-id-to-release-actions',
+  async up(trx, db) {
+    const hasPolymorphicColumn = await trx.schema.hasColumn('strapi_release_actions', 'target_id');
+
+    if (hasPolymorphicColumn) {
+      const releaseActions = await trx.select('*').from('strapi_release_actions');
+
+      for (const releaseAction of releaseActions) {
+        const { target_type, target_id } = releaseAction;
+
+        const entry = await db.query(target_type).findOne({ where: { id: target_id } });
+
+        if (entry) {
+          await trx('strapi_release_actions')
+            .update({ entry_document_id: entry.documentId })
+            .where('id', releaseAction.id);
+        }
```

Comment:
Is the "entry_document_id" attribute added in this same PR? If it is, people running this migration for the first time might not have this field in their schema.

also, could this be batched somehow? Or at least make this concurrent with async.map so migration is a bit quicker.